tests/foreman/cli/test_proxy.py

```diff
@@ -97,16 +73,18 @@ def test_proxy_update(self, test_data):
         @Assert: Proxy has the name updated
 
         """
-        proxy = make_proxy({u'name': test_data['name']})
-        self.assertEquals(proxy['name'], test_data['name'])
-        with default_url_on_new_port(9090, random.randint(9091, 49090)) as url:
-            Proxy.update({
-                u'id': proxy['id'],
-                u'name': test_data['update'],
-                u'url': url,
-            })
-        proxy = Proxy.info({u'id': proxy['id']})
-        self.assertEqual(proxy['name'], test_data['update'])
+        proxy = make_proxy({u'name': gen_alphanumeric()})
+        newport = random.randint(9091, 49090)
+        for new_name in valid_data_list():
+            with self.subTest(new_name):
+                with default_url_on_new_port(9090, newport) as url:
```

Comment:
to make this context manager more readable and to make flake8 / pylint happy, a new variable called newport is introduced.